app/components/agenda/agenda-header/agenda-check.js

```diff
@@ -0,0 +1,55 @@
+import Component from '@glimmer/component';
+import { A } from '@ember/array';
+import { trackedTask } from 'reactiveweb/ember-concurrency';
+import { task } from 'ember-concurrency';
+
+/**
+ * @argument onSave
+ * @argument onCancel
+ */
+
+export default class AgendaHeaderAgendaCheck extends Component {
+  getFileNameMappings = task(async () => {
+    const res = await fetch(`/document-naming/agenda/${this.args.agenda.id}`);
+    const mappings = await res.json();
+    const mappingsMap = new Map(
+      mappings.map(({ uri, newTitle }) => [uri, newTitle])
+    );
+
+    return mappingsMap;
+  });
+
+  fileNameMappings = trackedTask(this, this.getFileNameMappings);
+
+  get notaGroups() {
```

Comment:
This basically duplicates/re-writes `groupAgendaitemsOnGroupName` in the agenda-service: https://github.com/kanselarij-vlaanderen/frontend-kaleidos/blob/development/app/services/agenda-service.js#L180-L213

This would cause us to have 2 places to maintain the same action, so I'd prefer if the existing one would be reused.